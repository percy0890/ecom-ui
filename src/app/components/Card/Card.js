import React from 'react';
import PropTypes from 'prop-types';
import IMG from 'app/utils/images';
import AppConstants from 'app/app.constants.json';
import './Card.scss';
import { naCheck } from 'app/utils/common';

class Card extends React.PureComponent {
  clickCard = () => {
    this.props.onCardExpand(this.props.id);
  };

  checkValidationOnSelection = () =>
    this.props.selectedCard &&
    this.props.selectedCard.includes(this.props.id) &&
    this.props.tapType === AppConstants.CLICK_TYPE.SELECT;

  getSelectedClass = () => {
    if (
      this.checkValidationOnSelection() &&
      this.props.shiftType === AppConstants.SHIFT_TYPE.LOGIN
    )
      return 'select-card-pickup';
    if (
      this.checkValidationOnSelection() &&
      this.props.shiftType === AppConstants.SHIFT_TYPE.LOGOUT
    )
      return 'select-card-drop';
    return '';
  };

  getShiftClass = () => {
    if (
      this.checkValidationOnSelection() &&
      this.props.shiftType === AppConstants.SHIFT_TYPE.LOGIN
    )
      return 'shifts-pickup';
    if (
      this.checkValidationOnSelection() &&
      this.props.shiftType === AppConstants.SHIFT_TYPE.LOGOUT
    )
      return 'shifts-drop';
    return '';
  };

  render() {
    return (
      <React.Fragment>
        {this.props.selectedCard &&
        this.props.selectedCard.includes(this.props.id) &&
        this.props.tapType === AppConstants.CLICK_TYPE.EXPAND ? (
          <div
            className="card-container-open"
            role="presentation"
            onClick={this.clickCard}
          >
            <div className={`upper-container ${this.props.customClass}`}>
              <div className="shift-type-block">
                {this.props.shiftIcon && (
                  <img
                    src={this.props.shiftIcon}
                    alt="dots"
                    className="image"
                  />
                )}
                <div className="shift-type">
                  {naCheck(
                    this.props.shiftType && this.props.shiftType.toUpperCase(),
                  )}
                </div>
              </div>
              <div className="shift-id-container">
                <div className="shift-id">
                  <span>SHIFT ID: </span>
                  <span className="text">{this.props.shiftId}</span>
                </div>
                {this.props.assignmentIcon && (
                  <img
                    src={this.props.assignmentIcon}
                    alt="dots"
                    className="image"
                  />
                )}
              </div>
            </div>
            <div className="bottom-container">
              <div className="booking-id">
                <span>BOOKING ID: </span>
                <span className="text">{naCheck(this.props.bookingId)}</span>
              </div>
              <div className="location-container">
                <div className="row-wrap">
                  <div className="date-time">
                    <div>{naCheck(this.props.startDate)}</div>
                    <div className="time">{naCheck(this.props.startTime)}</div>
                  </div>
                  <div className="address-box">
                    <div className="indicator green" />
                    <div className="text">
                      <span>{naCheck(this.props.startLocation)}</span>
                    </div>
                  </div>
                </div>
                <div className="row-wrap">
                  <div className="date-time">
                    <div>{naCheck(this.props.endDate)}</div>
                    <div className="time">{naCheck(this.props.endTime)}</div>
                  </div>
                  <div className="address-box">
                    <div className="indicator red" />
                    <div className="text">
                      <span>{naCheck(this.props.endLocation)}</span>
                    </div>
                  </div>
                </div>
              </div>
              <div className="info-container">
                <div className="info-row">
                  <div className="info-field">
                    <div className="label">Trip Status</div>
                    <div className="value">
                      {naCheck(this.props.tripStatus)}
                    </div>
                  </div>
                  <div className="info-field">
                    <div className="label">Vehicle No.</div>
                    <div className="value">
                      {naCheck(this.props.vehicleNumber)}
                    </div>
                  </div>
                </div>
                <div className="info-row">
                  <div className="info-field">
                    <div className="label">Driver Name</div>
                    <div className="value">
                      {naCheck(this.props.driverName)}
                    </div>
                  </div>
                  <div className="info-field">
                    <div className="label">Contact No.</div>
                    <div className="value">
                      {naCheck(this.props.contactNumber)}
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        ) : (
          <div
            className="card-container cursor-pointer"
            role="presentation"
            onClick={this.clickCard}
          >
            {this.props.selectedCard &&
            this.props.selectedCard.includes(this.props.id) &&
            this.props.tapType === AppConstants.CLICK_TYPE.SELECT ? (
              <div className={`left-container ${this.props.customClass}`}>
                <div className="title-block center-align">
                  <img
                    src={IMG.FILTER_RIDE}
                    alt="dots"
                    className="checked-image"
                  />
                </div>
              </div>
            ) : (
              <div className={`left-container ${this.props.customClass}`}>
                <div className="title-block">
                  {this.props.shiftIcon && (
                    <img
                      src={this.props.shiftIcon}
                      alt="dots"
                      className="image"
                    />
                  )}
                  <div className="title">
                    {naCheck(
                      this.props.shiftType &&
                        this.props.shiftType.toUpperCase(),
                    )}
                  </div>
                </div>
                <div className="description-block">
                  <div>{naCheck(this.props.startDate)}</div>
                  <div className="time">
                    {naCheck(this.props.startTime)}-
                    {naCheck(this.props.endTime)}
                  </div>
                </div>
              </div>
            )}
            <div className={`right-container ${this.getSelectedClass()} `}>
              <div className={`upper-right-container ${this.getShiftClass()} `}>
                <div className="shift-id-block">
                  <span>SHIFT ID: </span>
                  <span className="shift-id">
                    {naCheck(this.props.shiftId)}
                  </span>
                </div>
                {this.props.assignmentIcon && (
                  <img
                    src={this.props.assignmentIcon}
                    alt="dots"
                    className="shift-icon"
                  />
                )}
              </div>
              <div className="address-container">
                <div className="address-indicator-block">
                  {this.props.icon && <img src={this.props.icon} alt="dots" />}
                </div>
                <div className="address">
                  <span>
                    {naCheck(
                      this.props.shiftType === AppConstants.SHIFT_TYPE.LOGIN
                        ? this.props.startLocation
                        : this.props.endLocation,
                    )}
                  </span>
                </div>
              </div>
            </div>
          </div>
        )}
      </React.Fragment>
    );
  }
}

Card.propTypes = {
  id: PropTypes.any,
  customClass: PropTypes.string,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  startTime: PropTypes.string,
  endTime: PropTypes.string,
  shiftId: PropTypes.string,
  shiftType: PropTypes.string,
  icon: PropTypes.any,
  assignmentIcon: PropTypes.any,
  shiftIcon: PropTypes.any,
  onCardExpand: PropTypes.func,
  selectedCard: PropTypes.any,
  bookingId: PropTypes.string,
  contactNumber: PropTypes.string,
  driverName: PropTypes.string,
  vehicleNumber: PropTypes.string,
  tripStatus: PropTypes.string,
  startLocation: PropTypes.string,
  endLocation: PropTypes.string,
  tapType: PropTypes.string,
};

export default Card;
