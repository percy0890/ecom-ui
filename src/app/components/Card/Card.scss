@import '~app/styles/core';

.card-container {
  display: flex;
  background-color: $app-trip-card-bg;
  width: 46%;
  box-shadow: $app-shadow-container;
  border-radius: 0.5rem;
  margin: 2rem;

  .left-container {
    flex: 1;
    display: flex;
    flex-direction: column;
    color: $app-trip-card-bg;
    border-top-left-radius: inherit;
    border-bottom-left-radius: inherit;
    padding: 2.5rem 2rem;

    .title-block {
      display: flex;
      font-family: $app-font-bold;
      align-items: center;
      padding: 0.3rem 0;
      font-size: 1.4rem;

      &.center-align {
        justify-content: center;
      }

      .checked-image {
        width: 7rem;
      }

      .title {
        margin-left: 0.6rem;
      }

      .image {
        width: 2rem;
      }
    }

    .description-block {
      margin-top: 1.5rem;

      .time {
        font-family: $app-font-bold;
      }
    }

    &.logout-container {
      background-color: $app-logout-color;
    }

    &.login-container {
      background-color: $app-login-color;
    }
  }

  .right-container {
    flex: 3.2;
    display: flex;
    flex-direction: column;

    .shifts-pickup {
      background-color: $app-shift-pickup !important;
      border-radius: 0.8rem;
    }

    .shifts-drop {
      background-color: $app-shift-drop !important;
      border-radius: 0.8rem;
    }

    .upper-right-container {
      margin: 1rem 1.5rem;
      padding: 0.5rem 1rem;
      background-color: $app-container-color;
      display: flex;
      justify-content: space-between;
      border-radius: 0.6rem;

      .shift-id-block {
        font-family: $app-font-medium;

        .shift-id {
          font-family: $app-font-bold;
        }
      }

      .shift-icon {
        width: 2rem;
      }
    }

    .address-container {
      margin: 0 1.5rem 1rem;
      display: flex;

      .address-indicator-block {
        img {
          width: 1.1rem;
        }
      }

      .address {
        padding-left: 0.8rem;
        font-family: $app-font-regular;
      }
    }
  }

  .select-card-pickup {
    background-color: $app-drop-select;
  }

  .select-card-drop {
    background-color: $app-pickup-select;
  }
}

@include media(tablet up) {
  .card-container {
    width: 46%;
    margin: 0.75rem 0.75rem;
  }
}

@include media(mobile down) {
  .card-container {
    width: 100%;
    margin: 0.75rem 0;
  }
}

.card-container-open {
  background-color: $app-trip-card-bg;
  width: 46%;
  box-shadow: $app-shadow-container;
  border-radius: 0.5rem;
  margin: 2rem;

  .upper-container {
    display: flex;
    color: $app-trip-card-bg;
    border-top-left-radius: inherit;
    border-top-right-radius: inherit;
    padding: 1.5rem;
    font-size: 1.6rem;

    &.logout-container {
      background-color: $app-login-color;
    }

    &.login-container {
      background-color: $app-logout-color;
    }

    .shift-type-block {
      flex: 1;
      display: flex;
      font-family: $app-font-bold;
      align-items: center;

      .shift-type {
        flex: 8;
        margin: 0.2rem 1rem;
      }

      .image {
        width: 2rem;
      }
    }

    .shift-id-container {
      flex: 2;
      display: flex;
      align-items: center;
      justify-content: space-between;

      .shift-id {
        padding: 0 1rem;
        font-family: $app-font-regular;

        .text {
          font-family: $app-font-bold;
        }
      }

      .image {
        width: 2rem;
      }
    }
  }

  .bottom-container {
    padding: 1.5rem;
    font-family: $app-font-regular;

    .booking-id {
      font-family: $app-font-regular;
      padding: 1rem 0;

      .text {
        font-family: $app-font-bold;
      }
    }

    .location-container {
      overflow: hidden;

      .row-wrap {
        display: flex;
        margin: 1.5rem 0;
        font-family: $app-font-regular;

        .date-time {
          line-height: 2.2rem;
          flex-shrink: 0;
          padding-right: 2.5rem;

          .time {
            font-family: $app-font-bold;
            font-size: 1.8rem;
          }
        }

        .address-box {
          display: flex;
          flex-grow: 1;

          .indicator {
            width: 1.5rem;
            height: 1.5rem;
            border-radius: 50%;
            flex-shrink: 0;
            position: relative;

            &.green {
              background-color: #7BC141;

              &:after {
                content: '';
                width: 2px;
                border-right: 2px dotted $app-border-color;
                position: absolute;
                top: 1.5rem;
                left: calc(50% - 1px);
                height: 100px;
              }
            }

            &.red {
              background-color: #CE2B04;

              &:after {
                content: '';
                width: 1.5rem;
                background-color: #fff;
                position: absolute;
                top: 1.5rem;
                left: 0;
                height: 50px;
              }
            }
          }
        }

        .text {
          padding-left: 1rem;
          font-size: 1.8rem;
        }
      }
    }

    .info-container {
      .info-row {
        display: flex;
        justify-content: space-between;
        margin-bottom: 1rem;

        .info-field {
          width: 48%;
          line-height: 3rem;

          .label {
            color: $brand-color-primary;
            font-family: $app-font-bold;
          }

          .value {
            padding: 0.5rem 1rem;
            background-color: $app-bg-body;
            border-radius: 0.6rem;
          }
        }
      }
    }
  }
}

@include media(tablet up) {
  .card-container-open {
    width: 46%;
    margin: 0.75rem 0.75rem;
  }
}

@include media(mobile down) {
  .card-container-open {
    width: 100% !important;
    margin: 0.75rem 0rem !important;

    .image {
      width: 2rem !important;
    }

    .dotted-image {
      height: 62.8% !important;
    }
  }
}
