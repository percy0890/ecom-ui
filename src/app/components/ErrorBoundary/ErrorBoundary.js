import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import './ErrorBoundary.scss';
import IMG from 'app/utils/images';
import Button from '../Button';

export class ErrorBoundary extends PureComponent {
  constructor(props) {
    super(props);

    const { history } = props;
    history.listen(() => {
      if (this.state.hasError) {
        this.setState({
          hasError: false,
        });
      }
    });

    this.state = { hasError: false };
  }

  componentDidCatch() {
    this.setState({ hasError: true });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div className="error-page">
          <section id="not-found">
            <div id="title">
              <Button
                id="confirm-model"
                buttonType="secondary"
                customClass="goback-btn"
                onClick={this.props.history.goBack}
                label="Go to previous page."
              />
            </div>
            <div className="circles">
              <p>
                <img src={IMG.ERROR_ICON} alt="error-icon" />
                <small>Oops! - Something went wrong.</small> <br />
                Please try again, or contact the support team for assistance if
                required.
              </p>
              <span className="circle big" />
              <span className="circle med" />
              <span className="circle small" />
            </div>
          </section>
        </div>
      );
    }
    return this.props.children;
  }
}

export default withRouter(ErrorBoundary);

ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired,
  history: PropTypes.object,
};
