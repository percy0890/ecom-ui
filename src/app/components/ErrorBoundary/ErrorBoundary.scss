@import '~app/styles/core';

.error-page {
  width: 100vw;
  height: 100vh;
  background: #ddd;
  margin: 0;
  padding: 0;

  #title {
    font-size: 40px;
    margin-bottom: -40px;
    position: relative;
    color: $app-text-white;
    padding: 0 2rem;

    .goback-btn {
      font-size: 16px;
    }
  }

  .circles:after {
    content: '';
    display: inline-block;
    width: 100%;
    height: 100px;
    background: $app-bg-white;
    position: absolute;
    top: -50px;
    left: 0;
    transform: skewY(-4deg);
    -webkit-transform: skewY(-4deg);
  }

  .circles {
    background: $app-bg-white;
    text-align: center;
    position: relative;
    margin-top: 174px;
    display: flex;
    justify-content: center;
    box-shadow: inset -1px -4px 4px rgba(0, 0, 0, 0.2);
  }

  .circles p {
    font-family: $app-font-bold;
    font-size: 24px;
    color: $app-text-secondary;
    position: relative;
    z-index: 9;
    line-height: 100%;
    height: 430px;
    width: 60%;

    img {
      height: 20rem;
      width: 20rem;
    }

    small {
      margin-top: 30px;
      display: block;
      color: $app-color-error;
      font-size: 40px;
      line-height: 100%;
      vertical-align: top;
    }
  }

  .circles .circle.small {
    width: 140px;
    height: 140px;
    border-radius: 50%;
    background: #bbb;
    position: absolute;
    z-index: 1;
    top: 80px;
    left: 50%;
    animation: 7s smallmove infinite cubic-bezier(1, 0.22, 0.71, 0.98);
    -webkit-animation: 7s smallmove infinite cubic-bezier(1, 0.22, 0.71, 0.98);
    animation-delay: 1.2s;
    -webkit-animation-delay: 1.2s;
  }

  .circles .circle.med {
    width: 200px;
    height: 200px;
    border-radius: 50%;
    background: #ccc;
    position: absolute;
    z-index: 1;
    top: 0;
    left: 10%;
    animation: 7s medmove infinite cubic-bezier(0.32, 0.04, 0.15, 0.75);
    -webkit-animation: 7s medmove infinite cubic-bezier(0.32, 0.04, 0.15, 0.75);
    animation-delay: 0.4s;
    -webkit-animation-delay: 0.4s;
  }

  .circles .circle.big {
    width: 400px;
    height: 400px;
    border-radius: 50%;
    background: #ddd;
    position: absolute;
    z-index: 1;
    top: 200px;
    right: 0;
    animation: 8s bigmove infinite;
    -webkit-animation: 8s bigmove infinite;
    animation-delay: 3s;
    -webkit-animation-delay: 1s;
  }

  @-webkit-keyframes smallmove {
    0% {
      top: 10px;
      left: 45%;
      opacity: 1;
    }
    25% {
      top: 300px;
      left: 40%;
      opacity: 0.7;
    }
    50% {
      top: 240px;
      left: 55%;
      opacity: 0.4;
    }
    75% {
      top: 100px;
      left: 40%;
      opacity: 0.6;
    }
    100% {
      top: 10px;
      left: 45%;
      opacity: 1;
    }
  }
  @keyframes smallmove {
    0% {
      top: 10px;
      left: 45%;
      opacity: 1;
    }
    25% {
      top: 300px;
      left: 40%;
      opacity: 0.7;
    }
    50% {
      top: 240px;
      left: 55%;
      opacity: 0.4;
    }
    75% {
      top: 100px;
      left: 40%;
      opacity: 0.6;
    }
    100% {
      top: 10px;
      left: 45%;
      opacity: 1;
    }
  }

  @-webkit-keyframes medmove {
    0% {
      top: 0px;
      left: 20%;
      opacity: 1;
    }
    25% {
      top: 300px;
      left: 80%;
      opacity: 0.7;
    }
    50% {
      top: 240px;
      left: 55%;
      opacity: 0.4;
    }
    75% {
      top: 100px;
      left: 40%;
      opacity: 0.6;
    }
    100% {
      top: 0px;
      left: 20%;
      opacity: 1;
    }
  }

  @keyframes medmove {
    0% {
      top: 0px;
      left: 20%;
      opacity: 1;
    }
    25% {
      top: 300px;
      left: 80%;
      opacity: 0.7;
    }
    50% {
      top: 240px;
      left: 55%;
      opacity: 0.4;
    }
    75% {
      top: 100px;
      left: 40%;
      opacity: 0.6;
    }
    100% {
      top: 0px;
      left: 20%;
      opacity: 1;
    }
  }

  @-webkit-keyframes bigmove {
    0% {
      top: 0px;
      right: 4%;
      opacity: 0.5;
    }
    25% {
      top: 100px;
      right: 40%;
      opacity: 0.4;
    }
    50% {
      top: 240px;
      right: 45%;
      opacity: 0.8;
    }
    75% {
      top: 100px;
      right: 35%;
      opacity: 0.6;
    }
    100% {
      top: 0px;
      right: 4%;
      opacity: 0.5;
    }
  }
  @keyframes bigmove {
    0% {
      top: 0px;
      right: 4%;
      opacity: 0.5;
    }
    25% {
      top: 100px;
      right: 40%;
      opacity: 0.4;
    }
    50% {
      top: 240px;
      right: 45%;
      opacity: 0.8;
    }
    75% {
      top: 100px;
      right: 35%;
      opacity: 0.6;
    }
    100% {
      top: 0px;
      right: 4%;
      opacity: 0.5;
    }
  }
}
