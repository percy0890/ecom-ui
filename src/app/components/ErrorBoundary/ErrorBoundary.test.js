import React from 'react';
import { shallow } from 'enzyme';
import Card from '../Card';
import { ErrorBoundary } from './ErrorBoundary';

describe('ErrorBoundary Component', () => {
  let wrapper;
  const props = { history: { listen: jest.fn() } };
  beforeEach(() => {
    wrapper = shallow(
      <ErrorBoundary {...props}>
        <Card>App</Card>
      </ErrorBoundary>,
    );
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has the default error-state set to false', () => {
    expect(wrapper.instance().state.hasError).toBe(false);
  });

  it('renders the ErrorBoundary when error is caught', () => {
    wrapper.instance().setState({ hasError: true });
    expect(wrapper.instance().state.hasError).toBe(true);
  });

  it('componentDidCatch should be called on catching error', () => {
    const componentDidCatch = jest.spyOn(
      wrapper.instance(),
      'componentDidCatch',
    );
    const error = new Error('test');

    wrapper.find('Card').simulateError(error);
    expect(componentDidCatch).toHaveBeenCalled();
  });
});
