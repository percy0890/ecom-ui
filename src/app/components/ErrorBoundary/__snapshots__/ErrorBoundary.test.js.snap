// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component renders correctly 1`] = `
<Card>
  App
</Card>
`;
